src/Raven.Server/Documents/Patch/V8/ScriptRunner.cs

```diff
@@ -0,0 +1,1700 @@
+                        throw new InvalidOperationException("endsWith(text, contained) must be called with two string parameters");
+
+                    return engine.CreateValue(args[0].AsString.EndsWith(args[1].AsString, StringComparison.OrdinalIgnoreCase));
+                }
+                catch (Exception e) 
+                {
+                    return engine.CreateError(e.Message, JSValueType.ExecutionError);
+                }
+            }
+
+            private InternalHandle RegexV8(V8Engine engine, bool isConstructCall, InternalHandle self, params InternalHandle[] args) // callback
+            {
+                try
+                {
+                    if (args.Length != 2 || args[0].IsStringEx == false || args[1].IsStringEx == false)
+                        throw new InvalidOperationException("regex(text, regex) must be called with two string parameters");
+
+                    var regex = _regexCache.Get(args[1].AsString);
+
+                    return engine.CreateValue(regex.IsMatch(args[0].AsString));
+                }
+                catch (Exception e) 
+                {
+                    return engine.CreateError(e.Message, JSValueType.ExecutionError);
+                }
+            }
+
+            private InternalHandle ScalarToRawStringV8(V8Engine engine, bool isConstructCall, InternalHandle self2, params InternalHandle[] args)
+            {
+                throw new NotImplementedException("scalarToRawString(document, lambdaToField) has not been implemented yet"); // TODO parser is needed here
+
+                /*if (args.Length != 2)
+                    throw new InvalidOperationException("scalarToRawString(document, lambdaToField) may be called on with two parameters only");
+
+                var firstParam = args[0];
+                if (firstParam.IsBinder && firstParam.BoundObject is BlittableObjectInstance selfInstance)
+                {
+                    var secondParam = args[1];
+                    if (secondParam.IsObject && secondParam.Object is V8Function lambda) // Jint: is ScriptFunctionInstance lambda)
+                    {
+                        var functionAst = lambda.FunctionDeclaration;
+                        string propName = functionAst.TryGetFieldFromSimpleLambdaExpression();
+
+                        IBlittableObjectProperty existingValue = default;
+                        if (selfInstance.TryGetValue(propName, out existingValue) && existingValue != null)
+                        {
+                            if (existingValue.Changed)
+                            {
+                                return existingValue.ValueHandle.V8.Item;
+                            }
+                        }
+
+                        var propertyIndex = selfInstance.Blittable.GetPropertyIndex(propName);
+
+                        if (propertyIndex == -1)
+                        {
```

Comment:
???

Reply:
it is marked as TODO
parser is needed here which is not available as of now:
                        var functionAst = lambda.FunctionDeclaration;
                        string propName = functionAst.TryGetFieldFromSimpleLambdaExpression();